contrib/grpcplugins/action/artifactory-plugins-lib/artifactory-plugins-lib.go

```diff
@@ -133,7 +140,28 @@ func promoteRunResult(ctx context.Context, c *actionplugin.Common, artifactClien
 		}
 	}
 
-	grpcplugins.Successf(c, "%s Successfully promoted to %s", r.Name(), newPromotion.ToMaturity)
+	if hasBeenPromoted {
+		grpcplugins.Successf(c, "%s Successfully promoted to %s", r.Name(), newPromotion.ToMaturity)
+	} else {
+		// Check existing release  on artifact
+		artifactFullPathSplit := strings.Split(promotedArtifact, r.ArtifactManagerMetadata.Get("path"))
+		props, err := artifactClient.GetProperties(artifactFullPathSplit[0], r.ArtifactManagerMetadata.Get("path"))
+		if err != nil {
+			return promotedArtifact, errors.Errorf("unable to check file properties %s: %v", promotedArtifact, err)
+		}
+		releaseValues, has := props["release.name"]
+		if has && len(releaseValues) > 0 {
+			if releaseValues[0] != futureReleaseName {
+				return promotedArtifact, errors.Errorf("run result %s is already part of another release: %s ", promotedArtifact, releaseValues[0])
+			}
+			releaseVersion, has := props["release.version"]
+			if has && len(releaseVersion) > 0 {
+				if releaseVersion[0] != futureReleaseVersion {
+					return promotedArtifact, errors.Errorf("run result %s has already been release in version %s", promotedArtifact, releaseVersion[0])
```

Comment:
```suggestion
					return promotedArtifact, errors.Errorf("Run result %s has already been release in version %s", promotedArtifact, releaseVersion[0])
```

Reply:
We should write error with grpcplugins.Errorf to have a red message in the job output